testing/test_deploy.py

```diff
@@ -158,20 +158,49 @@ def setup(args):
   if args.cluster:
     util.configure_kubectl(args.project, args.zone, args.cluster)
 
-  apply_command = ["ks", "apply", "default", "-c", "kubeflow-core",]
-
-  util.run(apply_command, cwd=app_dir)
-
-  # Verify that the TfJob operator is actually deployed.
-  tf_job_deployment_name = "tf-job-operator"
-  logging.info("Verifying TfJob controller started.")
-  util.wait_for_deployment(api_client, namespace.metadata.name,
-                           tf_job_deployment_name)
-
-  # Verify that JupyterHub is actually deployed.
-  jupyter_name = "tf-hub"
-  logging.info("Verifying TfHub started.")
-  util.wait_for_statefulset(api_client, namespace.metadata.name, jupyter_name)
+  if args.deploy_core:
+    apply_command = ["ks", "apply", "default", "-c", "kubeflow-core",]
+
+    util.run(apply_command, cwd=app_dir)
+
+    # Verify that the TfJob operator is actually deployed.
+    tf_job_deployment_name = "tf-job-operator"
+    logging.info("Verifying TfJob controller started.")
+    util.wait_for_deployment(api_client, namespace.metadata.name,
+                             tf_job_deployment_name)
+
+    # Verify that JupyterHub is actually deployed.
+    jupyter_name = "tf-hub"
+    logging.info("Verifying TfHub started.")
+    util.wait_for_statefulset(api_client, namespace.metadata.name, jupyter_name)
+
+  if args.deploy_tf_serving:
+    logging.info("Deploying tf-serving.")
+    model_server_image = (args.model_server_image + "-" +
```

Comment:
Why not just take the model_server_image as an argument and do any substitution needed (e.g. add sha #) in ksonnet?

Reply:
jsonnet doesn't use ${SHA}.  You should take whatever parameters you need as workflow parameters and then use those in the name (or you could get them by parsing the prow_env). 

To start with I would use name since the name will be unique and something like {JOB_NAME}-{SHA}-{PR} etc...
So use that as the label for the image and truncate it if necessary.

Reply:
I was trying to use prow_env, just like "--github_token=$(GITHUB_TOKEN)"
But when I do: "--model_server_image=$(SERVING_IMAGE)-$(JOB_NAME)-$(SHA)",
$(JOB_NAME) and $(SHA) are not replaced by prow env var. So the python code is getting a string "gcr.io/xxx/:1.0-$(JOB_NAME)-$(SHA)", which is a invalid image.